modules/backend/behaviors/RelationController.php

```diff
@@ -1099,7 +1099,9 @@ public function onRelationManageCreate()
                 $modelToSave->save(null, $this->manageWidget->getSessionKey());
             }
 
-            $this->relationObject->add($newModel, $sessionKey);
+            if (!$this->model->{$this->relationName}->contains($newModel)) {
```

Comment:
This will load all related models from the database to check for duplicates.

If you have a large dataset, the process is likely to run out of memory.

Reply:
Allright. Do you have a suggestion for a fix? 😃

Reply:
From your issue description, you appear to be trying to solve a problem in your code at the framework level. Why are you saving the relationship in the afterSave event?

Reply:
Because the repeater should tell the parent where to link to

<img width="1111" alt="CleanShot 2019-12-13 at 15 30 57@2x" src="https://user-images.githubusercontent.com/6506510/70807310-98ee4d00-1dbd-11ea-930f-958f24abfe02.png">